network/network.go

```diff
@@ -198,9 +206,20 @@ func (c *Collector) Collect(ch chan<- prometheus.Metric) {
 			c.logger.Log("level", "info", "message", "dialing host", "host", host, "scrapeID", c.scrapeID)
 			conn, err := c.dialer.Dial("tcp", host)
 			if err != nil {
-				c.logger.Log("level", "error", "message", "could not dial host", "host", host, "scrapeID", c.scrapeID, "stack", microerror.Stack(err))
-				c.dialErrorCount.WithLabelValues(host).Inc()
-				return
+				podExists, err := c.podExists(host, pods)
+				if err != nil {
+					c.logger.Log("level", "error", "message", "unable to check if host exists", "host", host, "scrapeID", c.scrapeID, "stack", microerror.Stack(err))
+					return
+				}
+
+				if podExists {
+					c.logger.Log("level", "error", "message", "could not dial host", "host", host, "scrapeID", c.scrapeID, "stack", microerror.Stack(err))
```

Comment:
I think you wanted to log `err` from outer `if`. But this one is overridden by `c.podExists` call.

Reply:
You're right, good catch